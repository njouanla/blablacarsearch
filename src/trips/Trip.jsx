import React from 'react';
import PropTypes from 'prop-types';

export const propTypes = {
    tripData: PropTypes.shape({
        distance_in_meters: PropTypes.number,
        duration_in_seconds: PropTypes.number,
        link: PropTypes.string,
        price: PropTypes.shape({
            amount: PropTypes.string,
            currency: PropTypes.string
        }),
        waypoints: PropTypes.arrayOf(PropTypes.shape({
            date_time: PropTypes.string,
            place: PropTypes.shape({
                address: PropTypes.string,
                city: PropTypes.string,
                country_code: PropTypes.string,
                latitude: PropTypes.number,
                longitude: PropTypes.number
            })
        }))
    })
};

const Trip = ({tripData}) => {
    return (<div>
        <div>horaires + destinations</div>
        <div>prix</div>
        <div>conducteur</div>
    </div>)
};

Trip.propTypes = propTypes;
export default Trip;