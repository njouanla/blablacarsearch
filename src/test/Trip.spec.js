import React from 'react';
import renderer from 'react-test-renderer';
import Trip from "../trips/Trip";
import {mockedResponse} from "./mocks";

describe('Trip', () => {
    it('should render Trip', () => {
        const wrapper = renderer
            .create(<Trip tripData={mockedResponse.trips[0]}/>)
            .toJSON();
        expect(wrapper).toMatchSnapshot();
    });
});