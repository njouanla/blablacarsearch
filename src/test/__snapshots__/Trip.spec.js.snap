// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trip should render Trip 1`] = `
<div
  className="sc-eCssSg dNYkuu"
>
  <div
    className="sc-jSgupP cXdENn"
  >
    <div>
      <div
        className="sc-gKsewC bxFDn"
      >
        <div
          className="sc-iBPRYJ fIRetL"
        >
          14:30
        </div>
        <div
          className="sc-iBPRYJ sc-fubCfw fIRetL iBxxMk"
        >
          Paris
        </div>
      </div>
      <div
        className="sc-gKsewC bxFDn"
      >
        <div
          className="sc-iBPRYJ fIRetL"
        >
          19:20
        </div>
        <div
          className="sc-iBPRYJ sc-fubCfw fIRetL iBxxMk"
        >
          Lyon
        </div>
      </div>
    </div>
    <div
      className="sc-iBPRYJ sc-fubCfw fIRetL ksShyK"
    >
      37.00
      €
    </div>
  </div>
  <a
    href="https://www.blablacar.fr/trip?source=CARPOOLING&id=2090443026-paris-lyon"
  >
    Voir le détail
  </a>
</div>
`;
